plugins/inputs/logfile/tailersrc.go

```diff
@@ -273,24 +266,102 @@ func (ts *tailerSrc) runTail() {
 				cnt++
 			}
 
-			if cnt < 5 {
-				continue
+			if cnt >= 5 {
+				ts.publishEvent(msgBuf, fo)
+				msgBuf.Reset()
+				cnt = 0
+			}
+		case <-ts.done:
+			return
+		}
+	}
+}
+
+func (ts *tailerSrc) publishEvent(msgBuf bytes.Buffer, fo *fileOffset) {
+	// helper to handle event publishing
+	if msgBuf.Len() == 0 {
+		return
+	}
+	msg := msgBuf.String()
+	timestamp, modifiedMsg := ts.timestampFn(msg)
+	e := &LogEvent{
+		msg:    modifiedMsg,
+		t:      timestamp,
+		offset: *fo,
+		src:    ts,
+	}
+	if ShouldPublish(ts.group, ts.stream, ts.filters, e) {
+		if ts.backpressureDrop {
+			select {
+			case ts.buffer <- e:
+				// successfully sent
+			case <-ts.done:
+				return
+			default:
+				// Buffer is full, close file and try again
+				log.Printf("D! [logfile] tailer sender buffer is full, closing file %v", ts.tailer.Filename)
+				ts.tailer.CloseFile()
```

Comment:
There should be a delay between the buffer being detected as full and the file getting closed. Latency for a PLE request can vary depending on setup, so not having a delay could cause the agent to constantly close/reopen the file.